launch/docker.go

```diff
@@ -94,19 +99,41 @@ func (d *docker) runBuild(buildEntry buildEntry) error {
 	artVol := fmt.Sprintf("%s/:%s", hostArtDir, containerArtDir)
 	binVol := fmt.Sprintf("%s:%s", d.volume, "/opt/sd")
 	habVol := fmt.Sprintf("%s:%s", d.habVolume, "/opt/sd/hab")
+
+	// Overwrite steps for sd-local interact mode. The env will load later.
+	if d.interactMode {
+		buildEntry.Steps = []screwdriver.Step{
+			{
+				Name:    "sd-local-init",
+				Command: "env > /tmp/sd-local.env",
+			},
+		}
+	}
+
 	configJSON, err := json.Marshal(buildEntry)
 	if err != nil {
 		return err
 	}
 
 	logrus.Infof("Pulling docker image from %s...", buildImage)
-	err = d.execDockerCommand("pull", buildImage)
+	_, err = d.execDockerCommand("pull", buildImage)
 	if err != nil {
 		return fmt.Errorf("failed to pull user image %v", err)
 	}
 
 	dockerCommandArgs := []string{"container", "run"}
-	dockerCommandOptions := []string{"--rm", "-v", srcVol, "-v", artVol, "-v", binVol, "-v", habVol, buildImage, "/opt/sd/local_run.sh", string(configJSON), buildEntry.JobName, environment["SD_API_URL"], environment["SD_STORE_URL"], logfilePath}
+	dockerCommandOptions := []string{"--rm", "-v", srcVol, "-v", artVol, "-v", binVol, "-v", habVol, buildImage}
+	configJSONArg := string(configJSON)
+	if d.interactMode {
+		configJSONArg = fmt.Sprintf("'%s'", configJSONArg)
+	}
+	launchCommands := []string{"/opt/sd/local_run.sh", configJSONArg, buildEntry.JobName, environment["SD_API_URL"], environment["SD_STORE_URL"], logfilePath}
+	if d.interactMode {
+		dockerCommandOptions = append([]string{"-itd"}, dockerCommandOptions...)
+		dockerCommandOptions = append(dockerCommandOptions, "/bin/sh")
```

Comment:
[note] If it don't run sh in the foreground, it won't be able to attach and interactively run the shell later. We cannot run `local_run.sh` at this stage, we must run it after attaching the build container.